src/com/amazon/kinesis/streaming/agent/processing/processors/SingleLineDataConverter.java

```diff
@@ -11,43 +11,59 @@
  * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
  * See the License for the specific language governing permissions and limitations under the License.
  */
-package com.amazon.kinesis.streaming.agent.processing.processors;
 
-import java.nio.ByteBuffer;
-import java.nio.charset.StandardCharsets;
-
-import org.apache.commons.lang3.StringUtils;
+package com.amazon.kinesis.streaming.agent.processing.processors;
 
 import com.amazon.kinesis.streaming.agent.ByteBuffers;
 import com.amazon.kinesis.streaming.agent.processing.exceptions.DataConversionException;
 import com.amazon.kinesis.streaming.agent.processing.interfaces.IDataConverter;
+import com.google.common.base.Splitter;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
- * Replace the newline with a whitespace
- * Remove leading and trailing spaces for each line
- * 
- * Configuration looks like:
- * 
- * { "optionName": "SINGLELINE" }
- * 
- * @author chaocheq
+ * An implementation of {@link IDataConverter} that converts the data specified
+ * into a single line from (possibly) a multi-line representation and replaces
+ * the newline character with a whitespace. All leading and trailing spaces for
+ * a given line are removed.
+ * <p>
+ * This converter can be configured by setting {@code optionName} to {@code SINGLELINE}
+ * when configuring the agent. Example:
+ * <pre> {@code
+ *     ...
+ *     "dataProcessingOptions": [
+ *       {
+ *         "optionName": "SINGLELINE"
+ *       }
+ *     ]
+ *     ...
+ * } </pre>
  *
+ * @author chaocheq
+ * @author Willy Lulciuc
  */
 public class SingleLineDataConverter implements IDataConverter {
-    
+
+    /** Splits each line from the data specified by the newline character. */
+    private static final Splitter NEW_LINE_SPLITTER = Splitter.on(NEW_LINE).omitEmptyStrings().trimResults();
```

Comment:
The code behaves the same as before, only now we don't update the lines returned by `split()` in place (i.e. eliminating side effect).